pkg/rendering/renderer_alertmanager.go

```diff
@@ -83,6 +84,24 @@ func (r *Renderer) renderAlertManagerStatefulSet(res *resource.Resource) (*unstr
 		spec.Containers[0].Image = image
 	}
 
+	if util.ProxyEnvVarsAreSet() {
+		spec.Containers[0].Env = append(spec.Containers[0].Env, []corev1.EnvVar{
+			{
+				Name:  "HTTP_PROXY",
+				Value: os.Getenv("HTTP_PROXY"),
+			},
+			{
+				Name:  "HTTPS_PROXY",
+				Value: os.Getenv("HTTPS_PROXY"),
+			},
+			{
+				Name:  "NO_PROXY",
+				Value: os.Getenv("NO_PROXY"),
+			},
+		}...)
+		log.V(1).Info("Proxy configuration environment variables are set.", "HTTP_PROXY:", os.Getenv("HTTP_PROXY"), "HTTPS_PROXY:", os.Getenv("HTTPS_PROXY"), "NO_PROXY:", os.Getenv("NO_PROXY"))
```

Comment:
Do not need to set such environments for alertmanager. Alertmanager is using configuration which is defined in `alertmanager.yaml`. the user can update the alertmanager-config secret e.g.:
```
global:
  slack_api_url: xxxx
  http_config:
    proxy_url: http://54.237.28.209:3128
route:
  receiver: default-receiver
  group_by: ['alertname', 'cluster']
  repeat_interval: 30ms
  matchers:
  - severity="critical"
receivers:
```

Reply:
@clyang82 
imo, this is a bad user experience
can mco controller set it automatically based on the env?

Reply:
@clyang82 I agree with Marco. Asking the user to ALSO add this is not best. I agree with @marcolan018  - we should add this logic to modify to our MCO controller. BTW can we  check with Simon P on how the platform alertmanager behaves - just for reference.

Reply:
https://github.com/prometheus/alertmanager/issues/1300 the alertmanager stopped working with proxy environment. same as my test results
```
level=error ts=2021-07-02T02:29:46.469Z caller=dispatch.go:309 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="default-receiver/slack[0]: notify retry canceled after 2 attempts: Post \"<redacted>\": context deadline exceeded"
level=error ts=2021-07-02T02:30:53.882Z caller=dispatch.go:309 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="default-receiver/slack[0]: notify retry canceled after 2 attempts: Post \"<redacted>\": context deadline exceeded"
```

Reply:
@clyang82 
I mean mco controller to update alertmanger.yaml content based on the env. If PROXY_URL is available in env and no global.proxy_url set, then inject it.
Either to update the alertmanager-config secret directly, or generate an internal secret which based on it and mco operator env.

Reply:
OpenShift does not mention how to configure alertmanager via proxy - https://docs.openshift.com/container-platform/4.7/monitoring/managing-alerts.html#applying-custom-alertmanager-configuration_managing-alerts. I think we need to update document to tell the user how to configure. the good news is the user who configure the alertmanager should be the same user - IT admin who setup the ACM.

Reply:
@marcolan018 We do not configure any routes by default. if just update with `proxy_url`, it can be easy to overwrite by customer.

Reply:
@marcolan018 , @clyang82 and I had a long discussion with the Monitoring team here- https://coreos.slack.com/archives/C0VMT03S5/p1625236195269000?thread_ts=1625195650.257100&cid=C0VMT03S5
it seems that they have decided NOT to automatically use the proxy settings for the monitoring stack because all receivers for example may not need it. In other words it sounds like a design choice OpenShift has made consciously **because it does not work in all cases**. We can of course work to change this behavior - but that as you know will be long drawn process.

Reply:
@bjoydeep that make sense